cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocProvisioner.java

```diff
@@ -531,7 +562,8 @@ protected String getClusterName(ProvisionerContext context) throws Exception {
 
   private Optional<Cluster> findCluster(String clusterKey, DataprocClient client)
       throws RetryableProvisionException {
-    return client.getClusters(Collections.singletonMap(LABEL_RUN_KEY, clusterKey)).findAny();
+    return client.getClusters(Collections.singletonMap(LABEL_RUN_KEY, clusterKey),
+        new ErrorCategory(ErrorCategoryEnum.PROVISIONING)).findAny();
```

Comment:
How do you know it's provisioning? 
E.g. I see calls `deleteCluster`->`getRuntimeJobManager`->`getClusterName`->`findCluster`.
I think it would be hard and error prone to figure out error category on the spot. Also it looks more like some special stages of execution, not as error category. Error category may still be ACCESS / NETWORKING / etc. I would consider not deciding it here and wrapping in `WrappedStageException` or similar somewhere much higher (e.g. at ProvisioningSubtask level)

Reply:
I propagated `ErrorCategory` from `ClusterDeleteSubTask` to this method.